Extension/MultipathUpgrade/Model/AvailableResources.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.ViewModelCollection;
+using TaleWorlds.Core;
+
+namespace YAPO.MultipathUpgrade.Model
+{
+    public class AvailableResources
+    {
+        public AvailableResources(bool hotkeyControl)
+        {
+            int cost = 0;
+            if (!hotkeyControl)
+            {
+                int dailyCost = GetMostAccurateDailyCost();
+                cost = dailyCost * YapoSettings.Instance.DaysToPayDailyCostsBuffer;
+            }
+
+            AvailableGold = Hero.MainHero.Clan.Gold + cost;
+        }
+
+        public int AvailableGold { get; set; }
+        public Dictionary<ItemCategory, int> ItemsOfCategoryWithCount { get; set; } = new Dictionary<ItemCategory, int>();
+
+        private int GetMostAccurateDailyCost()
+        {
+            try
+            {
+                int.TryParse(
+                    CampaignUIHelper.GetGoldTooltip(Hero.MainHero.Clan)
+                        .First(property => property.DefinitionLabel == "Expected Change").ValueLabel, out int dailyChange);
+
+                return Math.Min(dailyChange, 0);
+            }
+            catch 
+            {
+                //ignored
+            }
+            return 0;
+        }
+
+        public void UpdateAvailableResources(PartyCharacterVM member, int troopsToUpgrade)
+        {
+            AvailableGold -= member.Character.UpgradeCost(PartyBase.MainParty, 0) * troopsToUpgrade;
+
+            if (member.Character.UpgradeRequiresItemFromCategory != null)
+            {
+                ItemsOfCategoryWithCount[member.Character.UpgradeRequiresItemFromCategory] -= troopsToUpgrade;
```

Comment:
What if there aren't enough items of this type to upgrade `troopsToUpgrade`?

Reply:
This is an update after the fact. However beforehand the resourcecost is determined in TroopActionService.CalculateCountOfTroopsThatCanBeUpgraded. That's where the itemcost is decided. This is to update that amount